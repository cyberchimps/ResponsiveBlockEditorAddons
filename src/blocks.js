//Section
import "./blocks/section/index.js";

//Advanced Columns
import "./blocks/advance-columns/index.js";
import "./blocks/advance-columns/column/index.js";

//Advanced Heading
import "./blocks/advanced-heading/index.js";

//Advanced Text
import "./blocks/advanced-text/index.js";

//Image
import "./blocks/image/index.js"

//Multi Buttons
import "./blocks/buttons/index.js";
import "./blocks/buttons/buttons-child/index.js";

//Call to Action
import "./blocks/call-to-action/index.js";

//Block Quote
import "./blocks/blockquote/index.js";

//Divider
import "./blocks/divider/index.js";

//Info Block
import "./blocks/info-block/index.js";

//Count Down
import "./blocks/count-down/index.js";

//Spacer
import "./blocks/spacer/index.js";

//Inline Notice
import "./blocks/inline-notice/index.js";

//Progress Bar
import "./blocks/progress-bar/index.js";

//Table of Contents
import "./blocks/table-of-contents/index.js";

//Testimonial
import "./blocks/testimonial/index.js";

//Count up
import "./blocks/count-up/index.js";

//Flipbox
import "./blocks/flipbox/index.js";

//Icons List
import "./blocks/icons-list/index.js";
import "./blocks/icons-list/icons-list-child/index.js";

//Google Map
import "./blocks/googlemap/index.js";

//Image Gallery
import "./blocks/gallery-masonry/index.js";

//Responsive Post and Page Grid
import "./blocks/post-grid/index.js";

//Post Carousel
import "./blocks/post-carousel/index.js";

//Post Timeline
import "./blocks/post-timeline/index.js";

//Image Boxes
import "./blocks/image-boxes/index.js";

//Shape Divider
import "./blocks/shape-divider/index.js";

//Accordian
import "./blocks/accordion/index.js";
import "./blocks/accordion/accordion-item/index.js";

//Content Timeline
import "./blocks/content-timeline/index.js";

//Image Slider
import "./blocks/image-slider/index.js";

//Team
import "./blocks/team/index.js";

//Expand/Show More
import "./blocks/expand/index.js";

//Card
import "./blocks/card/index.js";

//Pricing Table
import "./blocks/pricing-table/index.js";

//Pricing List
import "./blocks/pricing-list/index.js";

//Video Popup
import "./blocks/video-popup/index.js";

//Testimonial Slider
import "./blocks/testimonial-slider/index.js";

//Feature Grid
import "./blocks/feature-grid/index.js";

//Portfolio
import "./blocks/portfolio/index.js";

//Not to make available for users yet
// import "./blocks/how-to/index.js";

//Anchor
import "./blocks/anchor/index.js";

//Call/Mail Button
import "./blocks/call-mail-button/index.js";

//Social Icons
import "./blocks/social-share/index.js";

//Tabs
import "./blocks/tabs/index.js";
import "./blocks/tabs/tabs-child/index";

//Taxonomy List
import "./blocks/taxonomy-list/index.js";

//WP Search
import "./blocks/wp-search/index.js";

//Instagram
import "./blocks/instagram/index.js";

//Image Hotspot
import "./blocks/image-hotspot/index.js";

//Contact Form 7 Styler
import "./blocks/contact-form-7-styler/index.js";

//Popup
import "./blocks/popup/index.js";

//Form
import "./blocks/form/index.js";
import "./blocks/form/input/index.js";

//Section block
import "./blocks/section-block/index.js";
// Extensions
import "./extensions/cover-styles";
