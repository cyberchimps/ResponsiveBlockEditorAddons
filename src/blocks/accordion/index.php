<?php
/**
 * Server-side rendering for the accordion block.
 *
 * @since   1.0.0
 * @package Responsive Blocks
 */

/**
 * Accordion frontend assets funciton.
 *
 * @return void
 */
function responsive_block_editor_addons_accordion_add_frontend_assets() {
	$widget_blocks = get_option('widget_block');
	if ( has_block( 'responsive-block-editor-addons/accordion' ) ) {
		wp_enqueue_script(
			'responsive_block_editor_addons-accordion-front-script',
			plugins_url( 'accordion.js', __FILE__ ),
			array(),
			RESPONSIVE_BLOCK_EDITOR_ADDONS_VER,
			true
		);
	}
	else if (!empty($widget_blocks)) {
		foreach ($widget_blocks as $widget) {
			if (!empty($widget['content'])) {
				wp_enqueue_script(
					'responsive_block_editor_addons-accordion-front-script',
					plugins_url( 'accordion.js', __FILE__ ),
					array(),
					RESPONSIVE_BLOCK_EDITOR_ADDONS_VER,
					true
				);
			}
		}
	}
}

add_action( 'wp_enqueue_scripts', 'responsive_block_editor_addons_accordion_add_frontend_assets' );
