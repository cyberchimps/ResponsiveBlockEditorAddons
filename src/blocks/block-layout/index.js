/**
 * BLOCK: Pattern Importer Layout
 */

/**
 * Import dependencies.
 */
import Edit from "./components/edit";
import "./styles/style.scss";
import "./styles/styles.editor.scss";
import LayoutsProvider from "./components/layouts-provider";

/**
 * WordPress dependencies.
 */
const { __ } = wp.i18n;
const { registerBlockType } = wp.blocks;

/**
 * Register the block
 */
registerBlockType("responsive-block-editor-addons/rbea-templates", {
  title: __("Pattern Importer", "responsive-block-editor-addons"),
  description: __(
    "This allows you to add pre-designed blocks and layouts to your page.",
    "responsive-block-editor-addons"
  ),
  icon: "editor-table",
  category: "responsive-block-editor-addons",
  keywords: [
    __("patterns", "responsive-block-editor-addons"),
    __("importer", "responsive-block-editor-addons"),
    __("layouts", "responsive-block-editor-addons"),
  ],

  /* Render the block in editor. */
  edit: (props) => {
    return (
      <LayoutsProvider>
        <Edit {...props} />
      </LayoutsProvider>
    );
  },

  /* Save the block markup. */
  save: () => {
    return null;
  },
});

/**
 * Add a Pattern Importer button to the toolbar.
 */
let patternButtonExist = false;
wp.data.subscribe(() => {
  appendImportButton();
});

/**
 * Build the pattern importer button.
 */
function appendImportButton() {
  if (patternButtonExist) {
    return;
  }
  const toolbar = document.querySelector(".edit-post-header__toolbar");
  if (!toolbar) {
    return;
  }
  const buttonDiv = document.createElement("div");
  buttonDiv.classList.add("rbea-pattern-wrapper");
  let html = '<div class="rbea-import-pattern">';
  html += `<button id="patternButton" class="components-button components-icon-button" aria-label="${__(
    "Import Pattern",
    "responsive-block-editor-addons"
  )}"><i class="dashicons dashicons-editor-table rbea-import-pattern-button"></i> ${__(
    "RBEA Templates",
    "responsive-block-editor-addons"
  )}</button>`;
  html += "</div>";
  buttonDiv.innerHTML = html;
  toolbar.appendChild(buttonDiv);

  const innerToolbar = document.querySelector(
    ".components-accessible-toolbar.edit-post-header-toolbar"
  );
  if (innerToolbar) {
    innerToolbar.style.flexGrow = 0;
  }

  document
    .getElementById("patternButton")
    .addEventListener("click", rbeaRenderModalBlock);

  patternButtonExist = true;
}

/**
 * Render block (Modal) on click.
 */
function rbeaRenderModalBlock() {
  const block = wp.blocks.createBlock(
    "responsive-block-editor-addons/rbea-templates"
  );
  wp.data.dispatch("core/block-editor").insertBlocks(block);
}
