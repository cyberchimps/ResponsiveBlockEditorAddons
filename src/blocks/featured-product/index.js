/**
 * BLOCK: Featured Product Block
 */

//Import block dependencies and components
import Edit from "./components/edit";
import Save from "./components/save";
import attributes from "./attributes";

//Import Block Icon
import ResponsiveBlockEditorAddonsIcons from "../../block-icons";

//Import CSS
import "./styles/style.scss"
import "./styles/styles.editor.scss"

// Internationalization
const { __ } = wp.i18n;

//Register block
const { registerBlockType } = wp.blocks;


//Register the block
registerBlockType("responsive-block-editor-addons/featured-product", {
    title: __("Featured Product",
    "responsive-block-editor-addons"),
    description:__(
        "Visually highlight a product or variation and encourage prompt action.",
        "responsive-block-editor-addons"
    ),
    icon: ResponsiveBlockEditorAddonsIcons.featured_product,
    category: "responsive_block_editor_addons",
    keywords: [
        __("feature","responsive-block-editor-addons"),
        __("featured-product","responsive-block-editor-addons"),
        __("responsive", "responsive-block-editor-addons"),
    ],

    attributes: attributes,

    supports: {
        anchor: true,
        align: true,
        
    },

    /* Render the block in the editor. */
    edit: (props) => {
        return <Edit {...props} />;
    },

    /* Save the block markup. */
    save: (props) => {
        return <Save {...props}/>;
    },
}); 