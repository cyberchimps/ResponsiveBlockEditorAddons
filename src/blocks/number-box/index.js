/**
 * BLOCK: Responsive Blocks Number Box
 */

// Import block dependencies and components
import Edit from "./components/edit";
import Save from "./components/save";
import attributes from "./attributes";

//Import Block Icon
import ResponsiveBlockEditorAddonsIcons from "../../block-icons";

// Import CSS
import "./styles/style.scss";
import "./styles/styles.editor.scss";

// Internationalization
const { __ } = wp.i18n;

// Register block
const { registerBlockType } = wp.blocks;

// Register the block
registerBlockType("responsive-block-editor-addons/number-box", {
  title: __("Number Box", "responsive-block-editor-addons"),
  description: __(
    "Add an attractive style to number(s), which can be used to represent steps, process, highlights, services, procedures, etc.",
    "responsive-block-editor-addons"
  ),
  icon: ResponsiveBlockEditorAddonsIcons.advanced_heading,
  category: "responsive_block_editor_addons",
  keywords: [
    __("number", "responsive-block-editor-addons"),
    __("number box", "responsive-block-editor-addons"),
    __("responsive", "responsive-block-editor-addons"),
  ],

  attributes: attributes,

  supports: {
		anchor: true
	},

  /* Render the block in the editor. */
  edit: (props) => {
    return <Edit {...props} />;
  },

  /* Save the block markup. */
  save: (props) => {
    return <Save {...props} />;
  },
});
