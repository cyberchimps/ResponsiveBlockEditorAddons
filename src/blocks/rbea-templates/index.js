/**
 * BLOCK: Pattern Importer Layout
 */

/**
 * Import dependencies.
 */
import Edit from "./components/edit";
import "./styles/style.scss";
import "./styles/styles.editor.scss";
import LayoutsProvider from "./components/layouts-provider";

import ResponsiveBlockEditorAddonsIcons from "../../block-icons";

/**
 * WordPress dependencies.
 */
const { __ } = wp.i18n;
const { registerBlockType } = wp.blocks;

/**
 * Register the block
 */
registerBlockType("responsive-block-editor-addons/rbea-templates", {
  title: __("RBEA Templates", "responsive-block-editor-addons"),
  description: __(
    "This allows you to add pre-designed blocks and layouts to your page.",
    "responsive-block-editor-addons"
  ),
  icon: ResponsiveBlockEditorAddonsIcons.block_importer,
  category: "responsive-block-editor-addons",
  keywords: [
    __("patterns", "responsive-block-editor-addons"),
    __("importer", "responsive-block-editor-addons"),
    __("layouts", "responsive-block-editor-addons"),
  ],

  /* Render the block in editor. */
  edit: (props) => {
    return (
      <LayoutsProvider>
        <Edit {...props} />
      </LayoutsProvider>
    );
  },

  /* Save the block markup. */
  save: () => {
    return null;
  },
});

/**
 * Add a Pattern Importer button to the toolbar.
 */
let patternButtonExist = false;
wp.data.subscribe(() => {
  appendImportButton();
});

/**
 * Build the pattern importer button.
 */
function appendImportButton() {
  if (patternButtonExist) {
    return;
  }
  const toolbar = document.querySelector(".edit-post-header__toolbar");
  if (!toolbar) {
    return;
  }
  const buttonDiv = document.createElement("div");
  buttonDiv.classList.add("rbea-pattern-wrapper");
  let html = '<div class="rbea-import-pattern">';
  html += `<button id="patternButton" class="components-button components-icon-button" aria-label="${__(
    "Import Pattern",
    "responsive-block-editor-addons"
  )}"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" role="img" width="20" height="20" preserveAspectRatio="xMidYMid meet" viewBox="0 0 24 24"><g fill="#ff6f61"><path fill-rule="evenodd" clip-rule="evenodd" d="M14 7a1 1 0 0 0-1 1v8a1 1 0 0 0 1 1h4a1 1 0 0 0 1-1V8a1 1 0 0 0-1-1h-4zm3 2h-2v6h2V9z" fill="#ff6f61"/><path d="M6 7a1 1 0 0 0 0 2h4a1 1 0 1 0 0-2H6z" fill="#ff6f61"/><path d="M6 11a1 1 0 1 0 0 2h4a1 1 0 1 0 0-2H6z" fill="#ff6f61"/><path d="M5 16a1 1 0 0 1 1-1h4a1 1 0 1 1 0 2H6a1 1 0 0 1-1-1z" fill="#ff6f61"/><path fill-rule="evenodd" clip-rule="evenodd" d="M4 3a3 3 0 0 0-3 3v12a3 3 0 0 0 3 3h16a3 3 0 0 0 3-3V6a3 3 0 0 0-3-3H4zm16 2H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1V6a1 1 0 0 0-1-1z" fill="#ff6f61"/></g></svg> ${__(
    "RBEA Templates",
    "responsive-block-editor-addons"
  )}</button>`;
  html += "</div>";
  buttonDiv.innerHTML = html;
  toolbar.appendChild(buttonDiv);

  const innerToolbar = document.querySelector(
    ".components-accessible-toolbar.edit-post-header-toolbar"
  );
  if (innerToolbar) {
    innerToolbar.style.flexGrow = 0;
  }

  document
    .getElementById("patternButton")
    .addEventListener("click", rbeaRenderModalBlock);

  patternButtonExist = true;
}

/**
 * Render block (Modal) on click.
 */
function rbeaRenderModalBlock() {
  const block = wp.blocks.createBlock(
    "responsive-block-editor-addons/rbea-templates"
  );
  wp.data.dispatch("core/block-editor").insertBlocks(block);
}
