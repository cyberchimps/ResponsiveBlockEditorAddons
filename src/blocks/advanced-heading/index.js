/**
 * BLOCK: Responsive Blocks Advanced Heading
 */

// Import block dependencies and components
import Edit from "./components/edit";
import Save from "./components/save";
import attributes from "./attributes";

//Import Block Icon
import ResponsiveBlockEditorAddonsIcons from "../../block-icons";

// Import CSS
import "./styles/style.scss";
import "./styles/styles.editor.scss";

// Internationalization
// const { __ } = wp.i18n;

// Register block
// const { registerBlockType } = wp.blocks;

import {__} from "@wordpress/i18n";
import { registerBlockType } from "@wordpress/blocks";

const name = "responsive-block-editor-addons/advanced-heading";
const settings = {
  title: __("Advanced Heading", "responsive-block-editor-addons"),
  description: __(
    "Add a combination of a heading and a sub-heading with a separator in between.",
    "responsive-block-editor-addons"
  ),
  icon: ResponsiveBlockEditorAddonsIcons.advanced_heading,
  keywords: [
    __("heading", "responsive-block-editor-addons"),
    __("advanced heading", "responsive-block-editor-addons"),
    __("responsive", "responsive-block-editor-addons"),
  ],

  attributes,

  supports: {
		anchor: true
	},

  /* Render the block in the editor. */
  edit: (props) => {
    return <Edit {...props} />;
  },

  /* Save the block markup. */
  save: (props) => {
    return <Save {...props} />;
  },
};

// Register the block
registerBlockType("responsive-block-editor-addons/advanced-heading", {
  title: __("Advanced Heading", "responsive-block-editor-addons"),
  description: __(
    "Add a combination of a heading and a sub-heading with a separator in between.",
    "responsive-block-editor-addons"
  ),
  icon: ResponsiveBlockEditorAddonsIcons.advanced_heading,
  category: "responsive_block_editor_addons",
  keywords: [
    __("heading", "responsive-block-editor-addons"),
    __("advanced heading", "responsive-block-editor-addons"),
    __("responsive", "responsive-block-editor-addons"),
  ],

  attributes: attributes,

  supports: {
		anchor: true
	},

  /* Render the block in the editor. */
  edit: (props) => {
    return <Edit {...props} />;
  },

  /* Save the block markup. */
  save: (props) => {
    return <Save {...props} />;
  },
});

export {settings, name};