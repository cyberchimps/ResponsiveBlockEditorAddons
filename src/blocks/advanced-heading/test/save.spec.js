import '@testing-library/jest-dom';
import {registerBlockType, createBlock, serialize} from '@wordpress/blocks';
import { settings } from '../index';

let block;
let serializedBlock;

const name = "responsive-block-editor-addons/advanced-heading";
const category = "common";

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category, ...settings} );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name );

		// Reset the reused variables.
		serializedBlock = '';
	} );

	it( 'should render with content', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toMatchSnapshot();
	} );
} );