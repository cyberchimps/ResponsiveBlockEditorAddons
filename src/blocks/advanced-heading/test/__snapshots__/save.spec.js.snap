// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsive-block-editor-addons/advanced-heading should render with content 1`] = `
"<!-- wp:responsive-block-editor-addons/advanced-heading -->
<div class=\\"responsive-block-editor-addons-block-advanced-heading block-undefined\\"><h2 class=\\"responsive-heading-title-text\\"></h2><div class=\\"responsive-heading-seperator-wrap\\"><div class=\\"responsive-heading-seperator\\"></div></div><p class=\\"responsive-heading-desc-text\\"></p></div>
<!-- /wp:responsive-block-editor-addons/advanced-heading -->"
`;
